### PR TITLE
DEV-2133: py37 upgrade

### PR DESCRIPTION
**Pull request by ProfOak**

### CODE REVIEW

**Review by wrkhenddher — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -81,14 +81,12 @@ integrate:
   allow_failure: true
   parallel:
     matrix:
-      - PY_VER: '3.6'
-        REPO_NAME: [data-tools, esbuild, gdcapi, gdcbatch, inspector]
       - PY_VER: '3.7'
-        REPO_NAME: [gdcadmin, sheepdog]
+        REPO_NAME: [gdcadmin, gdcbatch, esbuild, sheepdog]
```
Comment (stilesj-uchicago) [inline .gitlab-ci.yml:85]: I don't think I've seen this before. Is this to trigger downstream builds?

**Review by stilesj-uchicago — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (qqiao2024) [reply .gitlab-ci.yml:85]: yes

File: tox.ini [hunk 2]
```diff
@@ -11,6 +11,8 @@ setenv =
 extras=
     dev
 install_command = python -m pip install {opts} {packages}
+allowlist_externals=
+    env
```
Comment (qqiao2024) [inline tox.ini]: probably should remove this

**Review by qqiao2024 — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (ProfOak) [reply tox.ini]: Tox complains if programs are used in the `commands` sections that are external programs. I'm not certain if this shows up as an errored build but it makes the `env` command not work when it gets to the `commands` section. Should I remove `env` from that section as well?

Comment (qqiao2024) [reply tox.ini]: I think the env is good for debug. But keep our passwords in logs in pipeline might be a security concern?

**Pull request merged**